compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -753,8 +753,11 @@ private boolean isTupleTypeAssignableToArrayType(BTupleType source, BArrayType t
 
     private boolean isArrayTypeAssignableToTupleType(BArrayType source, BTupleType target,
                                                      List<TypePair> unresolvedTypes) {
-        if (source.state != BArrayState.UNSEALED
-                && (target.restType != null || target.tupleTypes.size() != source.size)) {
+        if (source.state == BArrayState.UNSEALED) {
+            if (target.restType == null) {
+                return false;
+            }
+        } else if (target.restType != null || target.tupleTypes.size() != source.size) {
```

Comment:
Closed array to a tuple with a rest type can pass in a scenario like the following right?

```ballerina
int[4] i = [1, 2, 3, 4];
[int, int, int...] j = i; // this is valid
```

I don't think we can simply return `false` if `target.restType != null`.

Reply:
Yes m'am. 🙏

Reply:
Fixed in cf480c6